test/base/src/jvmTest/kotlin/io/realm/test/compiler/EmbeddedTests.kt

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2022 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.realm.test.compiler
+
+import com.tschuchort.compiletesting.KotlinCompilation
+import com.tschuchort.compiletesting.SourceFile
+import io.realm.test.util.Compiler
+import org.junit.Test
+import kotlin.test.assertEquals
+import kotlin.test.assertTrue
+
+class EmbeddedTests {
```

Comment:
Missing tests for the cycle type fix similar to RealmObject

Reply:
I don't think they add any value, as verifying that we can handle the cyclic inner classes doesn't depend on whether we are using them together with `RealmObject` or `EmbeddedRealmObject` ... but added them to `CyclicDependenciesTests.kt` for completion 🤷 . I think the most important thing is to verify that we actually recognize all variations of `RealmObject` and `EmbeddedRealmObjects` so I added schema tests verifying that we also recognize both when using fully qualified class names instead of imports and added `FqNameImportTests` to verify that we can instantiate them to verify that all internal metadata is still correct. (This should already be verified numerous times for model definitions using normal imports)

Reply:
👍